hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -490,7 +490,7 @@ void onShutdownRequest(Address address) {
         }
 
         ClusterState clusterState = node.getClusterService().getClusterState();
-        if (clusterState == ClusterState.FROZEN || clusterState == ClusterState.PASSIVE) {
+        if (!clusterState.isMigrationAllowed()) {
```

Comment:
`NO_MIGRATION` should be included. 

But `IN_TRANSITION` should be excluded, right. Perfect finding! Will fix that.